steel/src/host/db/provider.rs

```diff
@@ -20,15 +20,29 @@ use alloy_primitives::{Address, BlockNumber, StorageKey};
 use anyhow::{ensure, Result};
 use revm::Database;
 
+#[derive(Clone, Debug)]
```

Comment:
I imagine we'll want to add fields to this in the future, so marking it non-exhaustive would make sense. 
```suggestion
#[derive(Clone, Debug)]
#[non_exhaustive]
```
Constructing it then means you need a builder pattern (or the dev can get the default and modify it as `mut`). One way to do that is the self-contained version like we use for `ProverOpts`, where each field has a `with_field` method that consumes `self` and returns the modified version.
https://github.com/risc0/risc0/blob/main/risc0/zkvm/src/host/client/prove/mod.rs#L288

Reply:
I agree, marking it `non_exhaustive` is definitely better.
Currently, since it has only one field, it can be set directly in the EnvBuilder: https://github.com/risc0/risc0-ethereum/blob/fdf72bb848d987cb7fc4050f25bfafec0d88c28e/steel/src/host/mod.rs#L168

But once we add more fields, I like the patter you mentioned.